demo/src/main/java/com/novoda/demo/MainActivity.java

```diff
@@ -58,22 +46,16 @@ protected void onCreate(Bundle savedInstanceState) {
                 .withDowngradedSecureDecoder()
                 .build(this);
 
-        player.getListeners().addPreparedListener(new NoPlayer.PreparedListener() {
-            @Override
-            public void onPrepared(PlayerState playerState) {
-                player.play();
-            }
-        });
-
-        player.attach(playerView);
+        demoPresenter = new DemoPresenter(controllerView, player, player.getListeners(), playerView);
+        dialogCreator = new DialogCreator(this, player);
     }
 
     @Override
     protected void onStart() {
         super.onStart();
         // TODO: Add switch in UI to avoid redeploy.
```

Comment:
If we're not going to do this we might as well drop the todo

Reply:
I have another branch where I have started to address this TODO. I'll drop it here in favour of opening an issue.

Reply:
672f1824628cd1b68325b1962db3c762ad4a918a